### ISSUE TITLE
Unclear changelogs

### ISSUE THREAD

**Issue by yogwoggf** [wiremod/wire#2662]
# Issue

Imagine being a server owner and wanting to see exactly what has changed in the latest wire update. Say one of your player complains about some bug, and you're not very sure how this bug happened. One obvious course of action is to look at the workshop changelog! The only place where Wiremod is packaged!
![image](https://github.com/wiremod/wire/assets/100450992/5f06c77c-d4e5-455c-97e9-107b1656d7ec)

That description provides no information to anyone involved.

# Potential solution

Use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) to easily give more information to each commit. Then, use [commitlint](https://commitlint.js.org/#/) and [husky](https://github.com/typicode/husky) to make sure each contributor properly names each commit.

Now that we have @Vurv78 's workshop deploy action, we can easily compile a list of commits since the last update and make a changelog that only consists of the `fix` and `feat` commits.

I'd also recommend ONLY pushing to the workshop for `fix` and `feat` commits since they will actually provide changes in how Wiremod works. That would also greatly reduce the amount of updates server owners have to deal with on each server boot.

**Comment by yogwoggf** [comment]
There's also an alarming amount of unclear commit names in Wire
![image](https://github.com/wiremod/wire/assets/100450992/4b16a1e8-2a8a-4464-9889-5ab2d57037df)
![image](https://github.com/wiremod/wire/assets/100450992/69ca5818-fa32-4f42-a308-7acc1be3b569)
![image](https://github.com/wiremod/wire/assets/100450992/4bfcdb18-4d32-43ab-bbb7-56c0ae0081ff)
![image](https://github.com/wiremod/wire/assets/100450992/850dfb2c-e253-4a9b-b3c2-ac773ed724b9)
![image](https://github.com/wiremod/wire/assets/100450992/43e609ef-fb6a-49ae-bbbd-0dadb93e719b)
![image](https://github.com/wiremod/wire/assets/100450992/fa230022-576d-4ea9-a579-63d25a70bc77)
![image](https://github.com/wiremod/wire/assets/100450992/3a97bb27-46f2-4a31-82ef-756eaeba3634)

**Comment by yogwoggf** [comment]
I also think that if server owners knew which commits caused a bug then triage of issues would be much faster.

**Comment by thegrb93** [comment]
Pinpointing bugs is what 'git bisect' is used for. You just have to describe a bug and when you think the bug started happening and git bisect and find the faulty commit very quickly.

Doing so by name of change is much slower.

**Comment by yogwoggf** [comment]
A server owner likely does not know what git bisect is, let alone have an installation of git. However, my suggestion will make `git bisect` much more efficient. We'd be able to tell a good commit from a bad commit from a simple glance at the changelogs if they were how I proposed them to be. If something bad starts happening after a update, then we know the last commit from the previous update is a good starting point by using the changelog.

**Comment by yogwoggf** [comment]
@Vurv78 @thegrb93 Any updates?

**Comment by Vurv78** [comment]
We'd have to stop squash merging to deal with this

**Comment by yogwoggf** [comment]
That doesn't seem like a bad thing. If we do implement the proposed solution then any bisecting would be benefited since the fine-grained commits will probably contain more specific changes (e.g 5 or so lines compared to a squashed commit which can be massive)

**Comment by thegrb93** [comment]
> We'd have to stop squash merging to deal with this

No, you could just replace the squash commit text with the PR title or something

**Comment by yogwoggf** [comment]
> > We'd have to stop squash merging to deal with this
> 
> No, you could just replace the squash commit text with the PR title or something

That would.. destroy the benefits of just merging a PR with a merge commit if we were to lint all commits, that way every wire update changelog would actually have every commit that did some change.

**Comment by thegrb93** [comment]
We don't want every commit someone's made in their branch in the baseline. A single commit with the main update they made should suffice.

**Comment by Anticept** [comment]
We had a tool that stopped working a long time ago that would grab commit information and compile it for updating the workshop. However, the person that maintained that system is long gone and we never got information on how he set it up.

Implementing such a system is welcome if someone wants to volunteer to do that. Until then, I'm not going to build all that by hand for the workshop; workshop always has been a secondary concern for wire.

**Comment by Aws0mee** [comment]
I personally don't see any issue with how it currently is. For my server I just use the github version and update it every so often or whenever I see an important change.
